terminal-ffm/src/main/java/org/jline/terminal/impl/ffm/FfmTerminalProvider.java

```diff
@@ -115,4 +119,13 @@ public int systemStreamWidth(SystemStream stream) {
     public String toString() {
         return "TerminalProvider[" + name() + "]";
     }
+
+    static VarHandle lookupVarHandle(MemoryLayout layout, PathElement... element) {
+        VarHandle h = layout.varHandle(element);
+
+        // the last parameter of the VarHandle is additional offset, hardcode zero:
+        h = MethodHandles.insertCoordinates(h, 1, 0L);
```

Comment:
Maybe do the following instead:
```suggestion
        h = MethodHandles.insertCoordinates(h, h.coordinateTypes().size() - 1, 0L);
```